### ISSUE TITLE
S4586 FP: when returning null from async lambda that has exactly 1 parameter without parentheses

### ISSUE THREAD

**Issue by antoinebj** [SonarSource/sonar-dotnet#2510]
### Description

False positive when returning null from async lambda that has exactly 1 parameter that is not surrounded by parentheses.

### Repro steps

```csharp
public class S4586
{
    public Task<object> SomeMethodAsync()
    {
        return this.SomeOtherMethodAsync(
            async input =>
            {
                await Task.Yield();

                return null; // S4586 here
            });
    }

    private Task<object> SomeOtherMethodAsync(Func<object, Task<object>> function)
    {
        return function.Invoke(default);
    }
}
```

### Expected behavior

No violation of rule

### Actual behavior

Violation of rule

### Known workarounds

Add parentheses around parameter:
```csharp
async (input) =>
```
(but this workaround might fail with automatic clean-up tools that remove the redundant parentheses)

**OR**

Return `default` instead of `null`:
```csharp
return default;
```

### Related information

* SonarC# Version 7.15
* Visual Studio Version 16.2
